fem/bilininteg.cpp

```diff
@@ -3458,17 +3464,9 @@ void DGDiffusionIntegrator::AssembleFaceMatrix(
    const IntegrationRule *ir = IntRule;
    if (ir == NULL)
    {
-      // a simple choice for the integration order; is this OK?
-      int order;
-      if (ndof2)
-      {
-         order = 2*max(el1.GetOrder(), el2.GetOrder());
-      }
-      else
-      {
-         order = 2*el1.GetOrder();
-      }
-      ir = &IntRules.Get(Trans.GetGeometryType(), order);
+      const int order = (ndof2) ? max(el1.GetOrder(),
+                                      el2.GetOrder()) : el1.GetOrder();
+      ir = &GetRule(order, Trans);
```

Comment:
What is motivating this change? Also, I think `GetRule` should be in `bilininteg.cpp`, it is currently very hard to locate!

Reply:
ping @pazner @arotem3

Reply:
This change was made in reference to an earlier comment by @termi-official in consideration for possible p (or hp) adaptivity in the future. Does this answer the question @YohannDudouit ?

Reply:
IIRC then we have some HP baseline already functional, see https://github.com/mfem/mfem/pull/1423

Reply:
As of 98cfc1fd6, the exact same quadrature rules should be used before and after this PR (just factored out to the `GetRule` member function)